### PR TITLE
KubernetesMockServer should not read local `.kube/config`

### LINKED ISSUE #3064: Client used by KubernetesServer should not try to read kubeconfig
It seems that when running tests using `KubernetesServer`, the client created try to read the user's local kubeconfig, which results in a bunch of errors, warnings. It does not really prevent the tests from passing, but greatly confuse the user when looking at the console output.

I suppose it must possible to have the created client just use the passed data from the `KubernetesServer` class and not try to read the local kubeconfig?

### LINKED ISSUE #3152: Mock Tests failing in `kubernetes-tests/` regarding `authorization.openshift.io` resources
Not sure if anyone noticed, but I'm seeing consistent failures of these tests on master in `kubernetes-tests/` module:
```
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClusterRoleTest.delete:80 
Expecting value to be true but was false
[ERROR]   ClusterRoleTest.get:46 
Expecting actual not to be null
[ERROR]   OpenshiftRoleBindingTest.testPatchWithOnlySubjects:168 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleBindingTest.testPatchWithUserNamesAndGroupsAndNoSubjects:189 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleBindingTest.testPatchWithUserNamesAndGroupsAndOverwriteSubjects:211 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleBindingTest.testReplaceWithOnlySubjects:110 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleBindingTest.testReplaceWithUserNamesAndGroupsAndNoSubjects:128 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleBindingTest.testReplaceWithUserNamesAndGroupsAndOverwriteSubjects:148 expected: <RoleBinding(apiVersion=authorization.openshift.io/v1, groupNames=[testgroup], kind=RoleBinding, metadata=ObjectMeta(annotations=null, clusterName=null, creationTimestamp=null, deletionGracePeriodSeconds=null, deletionTimestamp=null, finalizers=[], generateName=null, generation=null, labels=null, managedFields=[], name=null, namespace=null, ownerReferences=[], resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), roleRef=null, subjects=[ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser1, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=User, name=testuser2, namespace=null, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=ServiceAccount, name=svcacct, namespace=test, resourceVersion=null, uid=null, additionalProperties={}), ObjectReference(apiVersion=null, fieldPath=null, kind=Group, name=testgroup, namespace=null, resourceVersion=null, uid=null, additionalProperties={})], userNames=[testuser1, testuser2, system:serviceaccount:test:svcacct], additionalProperties={})> but was: <null>
[ERROR]   OpenshiftRoleTest.testGet:86 expected: not <null>
[ERROR]   RoleBindingRestrictionTest.delete:80 
Expecting value to be true but was false
[ERROR]   RoleBindingRestrictionTest.get:46 
Expecting actual not to be null
[ERROR] Errors: 
[ERROR]   ClusterRoleTest.list:60 » KubernetesClient Failure executing: GET at: https://...
[ERROR]   OpenshiftRoleBindingTest.testCreateInline:224 » KubernetesClient Failure execu...
[ERROR]   OpenshiftRoleBindingTest.testCreateWithOnlySubjects:47 » KubernetesClient Fail...
[ERROR]   OpenshiftRoleBindingTest.testCreateWithUserNamesAndGroupsAndNoSubjects:66 » KubernetesClient
[ERROR]   OpenshiftRoleBindingTest.testCreateWithUserNamesAndGroupsAndOverwriteSubjects:82 » KubernetesClient
[ERROR]   OpenshiftRoleTest.testList:61 » KubernetesClient Failure executing: GET at: ht...
[ERROR]   ProjectTest.testCreateProjectAndRoleBindings:192 » KubernetesClient Failure ex...
[ERROR]   RoleBindingRestrictionTest.list:60 » KubernetesClient Failure executing: GET a...
[ERROR]   SubjectAccessReviewTest.createLocalResourceAccessReview:138 » KubernetesClient
[ERROR]   SubjectAccessReviewTest.createResourceAccessReview:116 » KubernetesClient Fail...
[ERROR]   SubjectAccessReviewTest.createSelfSubjectRulesReviews:166 » KubernetesClient F...
[ERROR]   SubjectAccessReviewTest.createSubjectRulesReviews:193 » KubernetesClient Failu...
[ERROR]   SubjectAccessReviewTest.testCreate:54 » KubernetesClient Failure executing: PO...
[ERROR]   SubjectAccessReviewTest.testCreateInLine:68 » KubernetesClient Failure executi...
[ERROR]   SubjectAccessReviewTest.testCreateLocal:81 » KubernetesClient Failure executin...
[ERROR]   SubjectAccessReviewTest.testCreateLocalInLine:98 » KubernetesClient Failure ex...
[INFO] 
[ERROR] Tests run: 766, Failures: 11, Errors: 16, Skipped: 4
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10:28 min
[INFO] Finished at: 2021-05-20T21:19:07+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M5:test (default-test) on project kubernetes-test: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/rohaan/go/src/github.com/fabric8io/kubernetes-client/kubernetes-tests/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

I've even tried deleting my repository and rebuilding from scratch but I'm still seeing these failures. Surprisingly, I'm not able to see these failures on CI

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #3152 
Fix #3064

KubernetesMockServer is using [ConfigBuilder](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/test/java/io/fabric8/kubernetes/server/mock/KubernetesMockServer.java#L77) which tries to autoconfigure
Config by reading it automatically from local kubeconfig. this results
in issues where some code might accidentally read token/context set in
KubernetesClient's config which is set by Config autoconfigure.

This PR explicitly sets token, contexts to null/empty so that mockServer
tests run independently of local Kubernetes config

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java [hunk 1]
```diff
@@ -88,12 +89,21 @@ public RecordedRequest getLastRequest() throws InterruptedException {
...
                 .withTlsVersions(TLS_1_2)
                 .withNamespace("test")
                 .build();
-        return new DefaultKubernetesClient(createHttpClientForMockServer(config), config);
+        config.setCurrentContext(null);
+        config.setContexts(Collections.emptyList());
+        config.setOauthToken(null);
+
+        return config;
     }
```
Comment (manusa) [inline kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java:108]: Why isn't this working?
```java
        Config config = new ConfigBuilder()
                .withMasterUrl(url("/"))
                .withTrustCerts(true)
                .withTlsVersions(TLS_1_2)
                .withNamespace("test")
                .withContexts(Collections.emptyList())
                .build();
        return new DefaultKubernetesClient(createHttpClientForMockServer(config), config);
```

**Review by oscerd — APPROVED** [review]

File: kubernetes-server-mock/src/test/java/io/fabric8/kubernetes/client/server/mock/MockServerKubeconfigTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+class MockServerKubeconfigTest {
+
+  @BeforeEach
+  void setKubeConfigProperty() {
+    System.setProperty("kubeconfig", Objects.requireNonNull(getClass().getResource("/testkubeconfig")).getFile());
+  }
+
+  @Test
+  void mockServerShouldPickTokenAndNameContextIfKubeConfigFound() {
```
Comment (manusa) [inline kubernetes-server-mock/src/test/java/io/fabric8/kubernetes/client/server/mock/MockServerKubeconfigTest.java:36]: Shouldn't this be?
```suggestion
  void mockServerShouldNotPickTokenAndNameContextIfKubeConfigFound() {
```

File: kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java (hunk 1 above)
Comment (rohanKanojia) [reply kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java:108]: `ConfigBuilder()` calls this:
```java
    public ConfigBuilder() {
        this(true);
    }

    public ConfigBuilder(java.lang.Boolean validationEnabled) {
        this(new Config(), validationEnabled);
    }
```

`Config()` enables autoConfigure
```java
  public Config() {
    this(!Utils.getSystemPropertyOrEnvVar(KUBERNETES_DISABLE_AUTO_CONFIG_SYSTEM_PROPERTY, false));
  }
```
let me try to use `ConfigBuilder(Config.empty())` instead

Comment (manusa) [reply kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java]: I'm not following, in your changes you are using `new ConfigBuilder()` anyway.
The explicit calls to the setters (`setCurrentContext`, etc.) with null values should be redundant as (hopefully) those variables should be initialized as null.

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**